File: server/gb_api/api_objects/video_show.py
from . import image, video
from .api_object import ApiObject, ApiObjectDefinition, ApiField


class VideoShowDefinition(ApiObjectDefinition):
    def __init__(self):
        item_name = 'video_show'
        collection_name = 'video_shows'
        type_id = 2340
        object_class = VideoShow
        fields = [
            # URL pointing to the video_show detail resource.
            ApiField('api_detail_url', str),
            # Brief summary of the video_show.
            ApiField('deck', str),
            # For use in single item api call for video_show.
            ApiField('guid', str),
            # Unique ID of the video_show.
            ApiField('id', int),
            # Title of the video_show.
            ApiField('title', str),
            # Editor ordering.
            ApiField('position', str),
            # Main image of the video_show.
            ApiField('image', image.Image),
            # Show logo.
            ApiField('logo', image.Image),
            # URL pointing to the video_show on Giant Bomb.
            ApiField('site_detail_url', str),
            # Is this show currently active
            ApiField('active', str),
            # Should this show be displayed in navigation menus
            ApiField('display_nav', str),
            # The latest episode of a video show. Overrides other sorts when used as a sort field.
            ApiField('latest', video.Video),
            # Premium status of video_show.
            ApiField('premium', str),
            # Endpoint to retrieve the videos attached to this video_show.
            ApiField('api_videos_url', str)
        ]
        super().__init__(item_name, collection_name, type_id, object_class, fields)


class VideoShow(ApiObject):
    __definition = None

    def __init__(self, result):
        if self.__definition is None:
            self.__definition = VideoShowDefinition()
        super().__init__(self.__definition, result)

File: server/app/core/ts/VideoShow.ts
import Loadable from "./Loadable";
import {ImageResponseData} from "./gbmmapi/ImagesAPI";
import API, {validResponseData} from "./gbmmapi/API";
import {VideoShowResponseData} from "./gbmmapi/VideoShowsAPI";
import ResultList from "./ResultList";

export default class VideoShow extends Loadable {
    /** URL pointing to the detail resource. */
    apiDetailUrl: string
    /** Brief summary of the video_show. */
    deck: string
    /** Cross-entity unique ID */
    guid: string
    /** Title of the video_show. */
    title: string
    /** Editor ordering. */
    position: string
    /** Unique ID of the show. */
    id: number
    /** ID of the main image of the video_show. */
    imageId: number
    /** Main image of the video_show. */
    image?: ImageResponseData
    /** ID of the show logo. */
    logoId: number
    /** Show logo. */
    logo?: ImageResponseData
    /** Is this show currently active */
    active: boolean
    /** Should this show be displayed in navigation menus */
    displayNav: string
    /** The latest episode of a video show. Overrides other sorts when used as a sort field. */
    // latest
    /** Premium status of video_show. */
    premium: boolean
    /** Endpoint to retrieve the videos attached to this video_show. */
    apiVideosUrl: string
    /** URL pointing to the show on Giant Bomb. */
    siteDetailUrl: string
    /** Videos associated with this show. */
    // videos?: Video[]

    public constructor(data?: VideoShowResponseData) {
        super();
        if (!validResponseData(data)) {
            this.loadFailed = true;
            return;
        }
        this.apiDetailUrl = data.api_detail_url;
        this.deck = data.deck;
        this.guid = data.guid;
        this.title = data.title;
        this.position = data.position;
        this.id = data.id;
        this.imageId = data.image_id;
        this.image = data.image ?? null;
        this.logoId = data.logo_id;
        this.logo = data.logo ?? null;
        this.active = data.active;
        this.displayNav = data.display_nav;
        this.premium = data.premium;
        this.apiVideosUrl = data.api_videos_url;
        this.siteDetailUrl = data.site_detail_url;
    }

    public static async getAll() {
        let response = await API.videoShows.getAll();
        return new ResultList(VideoShow, response.data);
    }
}